pkg/status/combinedstatus-resolution.go

```diff
@@ -334,21 +338,54 @@ func (c *combinedStatusResolution) evaluateWorkStatus(celEvaluator *celEvaluator
 	}
 
 	updated := false
-	for statusCollectorName, scData := range c.statusCollectorNameToData {
-		changed, err := evaluateWorkStatusAgainstStatusCollectorWriteLocked(celEvaluator, workStatusWECName,
+	for _, scData := range c.statusCollectorNameToData {
+		changed := evaluateWorkStatusAgainstStatusCollectorWriteLocked(celEvaluator, workStatusWECName,
 			workStatusContent, scData)
-		if err != nil {
-			runtime2.HandleError(fmt.Errorf("failed to evaluate workstatus (%s) against statuscollector (%s): %w",
-				workStatusWECName, statusCollectorName, err))
-			continue
-		}
 
 		updated = updated || changed
 	}
 
 	return updated
 }
 
+func (c *combinedStatusResolution) requiresSourceObjectMetaOrSpec() bool {
+	c.RLock()
+	defer c.RUnlock()
+
+	pred := func(s string) bool {
+		return strings.Contains(s, fmt.Sprintf("%s.metadata", expressionRootName)) ||
+			strings.Contains(s, fmt.Sprintf("%s.spec", expressionRootName)) ||
+			strings.Contains(s, fmt.Sprintf("%s.kind", expressionRootName)) ||
```

Comment:
What if the expression contains `otherobj.kindly`?
What if the expression contains `someField.obj.kind`?
What if the expression contains `"some string about obj.kind"`?

Reply:
We can attach the source object's subresources in all cases. This here attempts to do so when we know that we **may** need them, and be sure not to fetch the object when we know it wont be needed.